core/templates/dev/head/pages/exploration_editor/editor_tab/ExplorationGraphDirective.ts

```diff
@@ -0,0 +1,124 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controllers for the exploration graph.
+ */
+
+require('domain/utilities/UrlInterpolationService.ts');
+require('pages/exploration_editor/ExplorationStatesService.ts');
+require('pages/exploration_editor/GraphDataService.ts');
+require('pages/exploration_editor/RouterService.ts');
+require('pages/state_editor/state_properties/StateEditorService.ts');
+require('services/AlertsService.ts');
+require('services/EditabilityService.ts');
+
+oppia.directive('explorationGraph', ['UrlInterpolationService', function(
+    UrlInterpolationService) {
+  return {
+    restrict: 'E',
+    scope: {},
+    bindToController: {},
+    templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+      '/pages/exploration_editor/editor_tab/' +
+      'exploration_graph_directive.html'),
+    controllerAs: '$ctrl',
+    controller: [
+      '$uibModal', 'AlertsService', 'EditabilityService',
+      'ExplorationStatesService', 'GraphDataService', 'RouterService',
+      'StateEditorService', 'UrlInterpolationService',
+      function(
+          $uibModal, AlertsService, EditabilityService,
+          ExplorationStatesService, GraphDataService, RouterService,
+          StateEditorService, UrlInterpolationService) {
+        var ctrl = this;
+        ctrl.getGraphData = GraphDataService.getGraphData;
+        ctrl.isEditable = EditabilityService.isEditable;
+
+        // We hide the graph at the outset in order not to confuse new
+        // exploration creators.
+        ctrl.isGraphShown = function() {
+          return Boolean(ExplorationStatesService.isInitialized() &&
+            ExplorationStatesService.getStateNames().length > 1);
+        };
+
+        ctrl.deleteState = function(deleteStateName) {
+          ExplorationStatesService.deleteState(deleteStateName);
+        };
+
+        ctrl.onClickStateInMinimap = function(stateName) {
+          RouterService.navigateToMainTab(stateName);
+        };
+
+        ctrl.getActiveStateName = function() {
+          return StateEditorService.getActiveStateName();
+        };
+
+        ctrl.openStateGraphModal = function() {
+          AlertsService.clearWarnings();
+
+          $uibModal.open({
+            templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+              '/pages/exploration_editor/editor_tab/' +
+              'exploration_graph_modal_directive.html'),
+            backdrop: true,
+            resolve: {
+              isEditable: function() {
+                return ctrl.isEditable;
+              }
+            },
+            windowClass: 'oppia-large-modal-window',
+            controller: [
```

Comment:
Do we also have to use `bindToController` in the modal? (not sure!)

Reply:
The main focus of this PR was to convert the controller part into directive. The modal is already a directive and I think we dont need to add `bindToController` here as of now in this PR.

Reply:
I see, thanks for the clarification!